src/analytics/mod.rs

```diff
@@ -343,42 +346,44 @@ impl<'a, I: InputSource> Slot<'a, I> {
         metadata: &TransactionMetadata,
         ctx: &BasicContext<'_>,
     ) -> eyre::Result<()> {
-        let consumed = payload
-            .transaction()
-            .inputs()
-            .iter()
-            .map(|input| input.as_utxo().output_id())
-            .map(|output_id| {
-                Ok(self
-                    .ledger_updates()
-                    .get_consumed(output_id)
-                    .ok_or(AnalyticsError::MissingLedgerSpent {
-                        output_id: *output_id,
-                        slot_index: metadata.transaction_id.slot_index(),
-                    })?
-                    .clone())
-            })
-            .collect::<eyre::Result<Vec<_>>>()?;
-        let created = payload
-            .transaction()
-            .outputs()
-            .iter()
-            .enumerate()
-            .map(|(index, _)| {
-                let output_id = metadata.transaction_id.into_output_id(index as _);
-                Ok(self
-                    .ledger_updates()
-                    .get_created(&output_id)
-                    .ok_or(AnalyticsError::MissingLedgerOutput {
-                        output_id,
-                        slot_index: metadata.transaction_id.slot_index(),
-                    })?
-                    .clone())
-            })
-            .collect::<eyre::Result<Vec<_>>>()?;
-        analytics
-            .handle_transaction(payload, metadata, &consumed, &created, ctx)
-            .await?;
+        if let Some(TransactionState::Finalized) = metadata.transaction_state {
```

Comment:
I think I'd prefer this just because there's no bindings in the match

```suggestion
        if metadata.transaction_state == Some(TransactionState::Finalized) {
```

Reply:
Also we should move the check up to the call-site so we don't have to indent this whole fn

Reply:
alrighty